src/main/scala/com/typesafe/sbt/packager/graalvmnativeimage/GraalVMNativeImagePlugin.scala

```diff
@@ -0,0 +1,191 @@
+    }
+    sys.process.Process(command, targetDirectory) ! log match {
+      case 0 => targetDirectory / binaryName
+      case x => sys.error(s"Failed to run $command, exit status: " + x)
+    }
+  }
+
+  private def buildInDockerContainer(targetDirectory: File,
+                                     binaryName: String,
+                                     className: String,
+                                     classpathJars: Seq[(File, String)],
+                                     extraOptions: Seq[String],
+                                     dockerCommand: Seq[String],
+                                     resources: Seq[(File, String)],
+                                     image: String,
+                                     streams: TaskStreams): File = {
+
+    stage(targetDirectory, classpathJars, resources, streams)
+
+    val command = dockerCommand ++ Seq(
+      "run",
+      "--rm",
+      "-v",
+      s"${targetDirectory.getAbsolutePath}:/opt/graalvm",
+      image,
+      "-cp",
+      classpathJars.map(jar => "/opt/graalvm/stage/" + jar._2).mkString(":"),
+      s"-H:Name=$binaryName"
+    ) ++ extraOptions ++ Seq(className)
+
+    sys.process.Process(command) ! streams.log match {
+      case 0 => targetDirectory / binaryName
+      case x => sys.error(s"Failed to run $command, exit status: " + x)
+    }
+  }
+
+  /**
+    * This can be used to build a custom build image starting from a custom base image. Can be used like so:
+    *
+    * ```
+    * (containerBuildImage in GraalVMNativeImage) := generateContainerBuildImage("my-docker-hub-username/my-graalvm").value
+    * ```
+    *
+    * The passed in docker image must have GraalVM installed and on the PATH, including the gu utility.
+    */
+  def generateContainerBuildImage(baseImage: String): Def.Initialize[Task[Option[String]]] = Def.task {
+    val dockerCommand = (DockerPlugin.autoImport.dockerExecCommand in GraalVMNativeImage).value
+    val streams = Keys.streams.value
+    val target = (Keys.target in GraalVMNativeImage).value
+
+    val (baseName, tag) = baseImage.split(":", 2) match {
+      case Array(n, t) => (n, t)
+      case Array(n)    => (n, "latest")
+    }
+
+    val imageName = s"${baseName.replace('/', '-')}-native-image:$tag"
+    import sys.process._
+    if ((dockerCommand ++ Seq("image", "ls", imageName, "--quiet")).!!.trim.isEmpty) {
+      streams.log.info(s"Generating new GraalVM native-image image based on $baseImage: $imageName")
+
+      val dockerFile = target / "Dockerfile.graalvm-native-image"
+
+      val dockerContent = Dockerfile(
+        Cmd("FROM", baseImage),
+        Cmd("WORKDIR", "/opt/graalvm"),
+        ExecCmd("RUN", "gu", "install", "native-image"),
+        ExecCmd("ENTRYPOINT", "native-image")
+      ).makeContent
+
+      IO.write(dockerFile, dockerContent)
+
+      DockerPlugin.publishLocalDocker(
+        target,
+        dockerCommand ++ Seq("build", "-f", dockerFile.getAbsolutePath, "-t", imageName, "."),
```

Comment:
This sends the whole `target` as a context, but doesn't use it... I don't have Docker handy at the moment, but something like that might work:
```
docker build -f <dockerFile> -t <imageName> -
```
You can also avoid an unnecessary `Dockerfile` in your target, and instead pipe that data directly into
```
docker build -t <imageName> -
```
though there's no way to do this with `publishLocalDocker` as it is now.

Reply:
Update: `docker build -f <dockerFile> -t <imageName> -` doesn't work for me: you only can build without context when you're piping the `Dockerfile`.

Reply:
Updated, this now uses `docker build -t <imageName> -`.

Reply:
Yep, I'm piping the docker file.